examples/platform/silabs/SiWx917/BUILD.gn

```diff
@@ -193,10 +193,10 @@ source_set("siwx917-common") {
     "${silabs_common_plat_dir}/SoftwareFaultReports.cpp",
     "${silabs_common_plat_dir}/silabs_utils.cpp",
     "${silabs_common_plat_dir}/syscalls_stubs.cpp",
-    "${silabs_plat_si91x_wifi_dir}/dhcp_client.cpp",
-    "${silabs_plat_si91x_wifi_dir}/ethernetif.cpp",
-    "${silabs_plat_si91x_wifi_dir}/lwip_netif.cpp",
-    "${silabs_plat_si91x_wifi_dir}/wfx_notify.cpp",
+    "${silabs_plat_si91x_wifi_dir}/SiWx917/wifi/ethernetif.cpp",
```

Comment:
Add a todo to merge the remaining to wifi folder as well

Reply:
@chirag-silabs Added